src/Server/TCPHandler.cpp

```diff
@@ -980,7 +981,7 @@ AsynchronousInsertQueue::PushResult TCPHandler::processAsyncInsertQuery(Asynchro
     Chunk result_chunk = Squashing::squash(squashing.flush());
     if (!result_chunk)
     {
-        return insert_queue.pushQueryWithBlock(state.parsed_query, squashing.getHeader(), query_context);
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Oh, wise CI gods, please help me find out how this could happen?!");
```

Comment:
```suggestion
        throw Exception(ErrorCodes::LOGICAL_ERROR, "Oh, Wise CI God, please help me find out how this could happen?!");
```

There is only one God. Down with the heretics of the Green CI. Red CI is the way.